core/Updates/3.6.0-b4.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Updates;
+
+use Piwik\Common;
+use Piwik\Date;
+use Piwik\Updater\Migration\Factory as MigrationFactory;
+use Piwik\Updates;
+use Piwik\Updater;
+
+/**
+ * Update for version 3.6.0-b4.
+ */
+class Updates_3_6_0_b4 extends Updates
+{
+    /**
+     * @var MigrationFactory
+     */
+    private $migration;
+
+    public function __construct(MigrationFactory $factory)
+    {
+        $this->migration = $factory;
+    }
+
+    public function getMigrations(Updater $updater)
+    {
+        // use php date since mysql date/time might be different and might lock users out for a while. and subtract days just to be safe.
+        $passwordModified = Date::factory('now')->subDay(14);
```

Comment:
the date-time will be 14 days offset from the actual time where password was changed, would it be possible not to have this 14 days offset somehow? As we may display in the future the time when the password was changed in the UI, the value should be accurate

Reply:
Then we shouldn't set the value on Matomo update at all and leave it blank. Guess the other code can be adjusted to ignore the additional checks if the password has never changed...

Setting the value to any other value, like `now()` or the user creation date might be wrong when displayed in the UI.

Reply:
Makes sense, will make the column null-able